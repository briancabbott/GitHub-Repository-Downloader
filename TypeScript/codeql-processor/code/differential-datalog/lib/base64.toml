[dependencies.base64]
version = "0.13"
