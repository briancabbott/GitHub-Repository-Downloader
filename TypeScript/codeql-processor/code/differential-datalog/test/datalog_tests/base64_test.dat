dump base64_test::Base64;
