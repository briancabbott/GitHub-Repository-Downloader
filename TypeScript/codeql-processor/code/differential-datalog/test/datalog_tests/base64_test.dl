import base64

output relation Base64(decoding: Result<Vec<u8>, string>, encoding: string, position: u8)

Base64(Ok{[0,1,2,3]}, encode([0,1,2,3]), 0).
Base64(decode(encode([0,1,2,3])), encode([0,1,2,3]), 1).
Base64(decode("TEST"), "TEST", 2).
Base64(decode("!@#!@#"), "!@#!@#", 3).
Base64(decode("Hello !@#!@#"), "Hello !@#!@#", 4).
