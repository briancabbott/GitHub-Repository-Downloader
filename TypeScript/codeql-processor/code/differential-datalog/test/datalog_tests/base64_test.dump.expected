base64_test::Base64{.decoding = ddlog_std::Ok{.res = [0, 1, 2, 3]}, .encoding = "AAECAw==", .position = 0}
base64_test::Base64{.decoding = ddlog_std::Ok{.res = [0, 1, 2, 3]}, .encoding = "AAECAw==", .position = 1}
base64_test::Base64{.decoding = ddlog_std::Ok{.res = [76, 68, 147]}, .encoding = "TEST", .position = 2}
base64_test::Base64{.decoding = ddlog_std::Err{.err = "Invalid byte 32 at position 5"}, .encoding = "Hello !@#!@#", .position = 4}
base64_test::Base64{.decoding = ddlog_std::Err{.err = "Invalid byte 33 at position 0"}, .encoding = "!@#!@#", .position = 3}
